lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1237 @@
+    assert len(enrollments) == 1, \
+        f"Record had {len(enrollments)} enrollments."
+
+    enrollment = enrollments[0]
+
+    incomplete_enrollment_instruments = {
+                instrument
+                    for instrument
+                    in REQUIRED_ENROLLMENT_INSTRUMENTS
+                    if not is_complete(instrument, enrollment)
+            }
+
+    if incomplete_enrollment_instruments:
+        LOG.debug(f"The following required enrollment instruments «{incomplete_enrollment_instruments}» are not yet marked complete.")
+        return None
+
+    # If the participant's age < 18 ensure we have parental consent.
+    if (enrollment['core_age_years'] == "" or int(enrollment['core_age_years']) < 18) and \
+            (is_complete('parental_consent_form', enrollment) == False or enrollment['signature_parent'] == ''):
+        LOG.debug("The participant is < 18 years old and we do not have parental consent. Skipping record.")
+        return None
+
+    patient_entry, patient_reference = create_patient(enrollment)
+    birthdate = datetime.strptime(enrollment['core_birthdate'], '%Y-%m-%d')
+
+    if not patient_entry:
+        LOG.warning("Skipping record with insufficient information to construct patient")
+        return None
+
+    location_resource_entries = locations(db, cache, enrollment)
+    persisted_resource_entries = [patient_entry, *location_resource_entries]
+
+    for redcap_record_instance in redcap_record_instances:
+
+        event_type = None
+        collection_method = None
+
+        if redcap_record_instance["redcap_event_name"] == ENROLLMENT_EVENT_NAME:
+            event_type = EventType.ENROLLMENT
+        elif redcap_record_instance["redcap_event_name"] == ENCOUNTER_EVENT_NAME:
+            event_type = EventType.ENCOUNTER
+        else:
+            LOG.error(f"The record instance has an unexpected event name: {redcap_record_instance['redcap_event_name']}")
+            continue
+
+        if event_type == EventType.ENCOUNTER:
+            if is_complete('kiosk_registration_4c7f', redcap_record_instance):
+                collection_method = CollectionMethod.KIOSK
+            elif is_complete('test_fulfillment_form', redcap_record_instance):
+                collection_method = CollectionMethod.SWAB_AND_SEND
+
+        # site_reference refers to where the sample was collected
+        site_reference = create_site_reference(redcap_record_instance, collection_method, event_type)
+
+        initial_encounter_entry, initial_encounter_reference = create_encounter(
+            redcap_record_instance, patient_reference, site_reference,
+            location_resource_entries, event_type, collection_method)
+
+        # Skip the entire record if we can't create the enrollment encounter.
+        # Otherwise, just skip the record instance.
+        if not initial_encounter_entry:
+            if event_type == EventType.ENROLLMENT:
+                return None
```

Comment:
I think we should also log a warning here if we have to skip an enrollment

Reply:
Added.